test/sqllogictest/ldbc_bi_eager.slt

```diff
@@ -2551,13 +2549,13 @@ Explained Query:
                                   Union // { arity: 2 }
                                     Negate // { arity: 2 }
                                       Project (#0{person1id}, #1{person2id}) // { arity: 2 }
-                                        Join on=(#2 = least(#0{person1id}, #1{person2id}) AND #3 = greatest(#0{person1id}, #1{person2id})) type=differential // { arity: 4 }
+                                        Join on=(#2 = greatest(#0{person1id}, #1{person2id}) AND #3 = least(#0{person1id}, #1{person2id})) type=differential // { arity: 4 }
                                           implementation
-                                            %1[#1, #0]UKK » %0:l2[greatest(#0, #1), least(#0, #1)]KK
+                                            %1[#0, #1]UKKA » %0:l2[greatest(#0, #1), least(#0, #1)]KK
                                           ArrangeBy keys=[[greatest(#0{person1id}, #1{person2id}), least(#0{person1id}, #1{person2id})]] // { arity: 2 }
                                             Get l2 // { arity: 2 }
-                                          ArrangeBy keys=[[#1, #0]] // { arity: 2 }
-                                            Distinct project=[#0, #1] // { arity: 2 }
+                                          ArrangeBy keys=[[#0, #1]] // { arity: 2 }
+                                            Distinct project=[#1, #0] // { arity: 2 }
```

Comment:
Something weird is going on here. The `A` sigil disappeared, meaning that `JoinImplementation` is no longer detecting an arrangement re-use. I'm confused, because in both the original and the new plan, the `ArrangeBy`'s key components are in the opposite order from the `Distinct`'s key components. Why can we detect a re-use in one case, but not the other?

Reply:
Ah, it's because it doesn't matter what is in the `Reduce`'s key. A `Reduce`'s output arrangement is always simply the first as many columns as the number of key components, in the `#0`, `#1`, `#2`, ... order.

Reply:
But I still don't understand why things got reordered here. Could a similar reordering happen in the other direction? If yes, then we have the potential to _lose_ an arrangement re-use in the same way as we gained one here?

Reply:
Turns out there is an issue for this: https://github.com/MaterializeInc/database-issues/issues/7505
Solution approaches mentioned there:
- One thought was to implement this in JoinImplementation: put all permutations (when the key is not too large) in available_arrangements when seeing a Reduce at an input, and then if JoinImplementation chooses one of these, then make that one actually happen by tweaking the Reduce. But this is not good, because a Delta join might use more than one, but we can only make one of the happen.
- Another thought would be to put a new transform in the JoinImplementation fixpoint loop, which would order the Reduce key components based on an already implemented join. But then it would be hard to 100% rule out infinite fixpoint loops.